cloud/misc/proxmox_template.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: proxmox_template
+short_description: management of OS templates in Proxmox VE cluster
+description:
+  - allows you to list/upload/delete templates in Proxmox VE cluster
+version_added: "2.0"
+options:
+  api_host:
+    description:
+      - the host of the Proxmox VE cluster
+    required: true
+  api_user:
+    description:
+      - the user to authenticate with
+    required: true
+  api_password:
+    description:
+      - the password to authenticate with
+      - you can use PROXMOX_PASSWORD environment variable
+    default: null
+    required: false
+  https_verify_ssl:
+    description:
+      - enable / disable https certificate verification
+    default: false
+    required: false
+    type: boolean
+  node:
+    description:
+      - Proxmox VE node, when you will operate with template
+    default: null
+    required: true
+  src:
+    description:
+      - path to uploaded file
+      - required only for C(state=present)
+    default: null
+    required: false
+    aliases: ['path']
+  template:
+    description:
+      - the template name
+      - required only for states C(absent), C(info)
+    default: null
+    required: false
+  content_type:
+    description:
+      - content type
+      - required only for C(state=present)
+    default: 'vztmpl'
+    required: false
+    choices: ['vztmpl', 'iso']
+  storage:
+    description:
+      - target storage
+    default: 'local'
+    required: false
+    type: string
+  timeout:
+    description:
+      - timeout for operations
+    default: 300
+    required: false
+    type: integer
+  force:
+    description:
+      - can be used only with C(state=present), exists template will be overwritten
+    default: false
+    required: false
+    type: boolean
+  state:
+    description:
+     - Indicate desired state of the template
+    choices: ['present', 'absent', 'list']
```

Comment:
list seems to indicate you want a prmox_facts kind of module also, there are some modules that do this in one place but we prefer to separate information from management modules